deps/exokit-bindings/magicleap/src/magicleap.cc

```diff
@@ -844,134 +940,248 @@ NAN_SETTER(MLPlaneTracker::OnPlanesSetter) {
+void setFingerValue(float *data);
+
+/* void setFingerValue(const MLWristState &wristState, MLSnapshot *snapshot, float data[4][1 + 3]) {
+  setFingerValue(wristState.radial, snapshot, data[0]);
+  setFingerValue(wristState.ulnar, snapshot, data[1]);
+  setFingerValue(wristState.center, snapshot, data[2]);
+  setFingerValue(data[3]);
+}
+void setFingerValue(const MLThumbState &thumbState, MLSnapshot *snapshot, float data[4][1 + 3]) {
+  setFingerValue(thumbState.cmc, snapshot, data[0]);
+  setFingerValue(thumbState.mcp, snapshot, data[1]);
+  setFingerValue(thumbState.ip, snapshot, data[2]);
+  setFingerValue(thumbState.tip, snapshot, data[3]);
+}
+void setFingerValue(const MLFingerState &fingerState, MLSnapshot *snapshot, float data[4][1 + 3]) {
+  setFingerValue(fingerState.mcp, snapshot, data[0]);
+  setFingerValue(fingerState.pip, snapshot, data[1]);
+  setFingerValue(fingerState.dip, snapshot, data[2]);
+  setFingerValue(fingerState.tip, snapshot, data[3]);
+} */
+void setFingerValue(const MLKeyPointState &keyPointState, MLSnapshot *snapshot, float *data) {
+  // uint32_t *uint32Data = (uint32_t *)data;
+  // float *floatData = (float *)data;
+
+  if (keyPointState.is_valid) {
+    MLTransform transform;
+    MLResult result = MLSnapshotGetTransform(snapshot, &keyPointState.frame_id, &transform);
+    if (result == MLResult_Ok) {
+      ML_LOG(Info, "%s: ML keypoint ok %f %f %f %x %x", application_name, transform.position.x, transform.position.y, transform.position.z, (void *)(&keyPointState.frame_id), (void *)data);
+
+      // uint32Data[0] = true;
+      data[0] = transform.position.x;
+      data[1] = transform.position.y;
+      data[2] = transform.position.z;
     } else {
-      return false;
+      // ML_LOG(Error, "%s: ML failed to get finger transform: %s", application_name, MLSnapshotGetResultString(result));
+
+      setFingerValue(data);
     }
-  }), planeTrackers.end());
+  } else {
+    setFingerValue(data);
+  }
 }
+void setFingerValue(float *data) {
+  // uint32_t *uint32Data = (uint32_t *)data;
+  // float *floatData = (float *)data;
 
-// MLHandTracker
+  // uint32Data[0] = false;
+  // const MLVec3f &position = MLVec3f{0, 0, 0};
+  auto nan = std::numeric_limits<float>::quiet_NaN();
+  data[0] = nan;
+  data[1] = nan;
+  data[2] = nan;
+}
 
-MLHandTracker::MLHandTracker(Local<Object> windowObj, uv_loop_t *loop) : windowObj(windowObj), loop(loop) {}
+MLHandTracker::MLHandTracker() {
+  if (MLHandTrackingCreate(&tracker) != MLResult_Ok) {
+    ML_LOG(Error, "%s: Failed to create hand tracker.", application_name);
+  }
+
+  MLHandTrackingConfiguration handTrackingConfig;
+  handTrackingConfig.handtracking_pipeline_enabled = true;
+  handTrackingConfig.keypoints_filter_level = MLKeypointFilterLevel_1;
+  handTrackingConfig.pose_filter_level = MLPoseFilterLevel_1;
+  for (int i = 0; i < MLHandTrackingKeyPose_Count; i++) {
+    handTrackingConfig.keypose_config[i] = true;
```

Comment:
Note for posterity: this flag set actually changes the way hand bones are detected. If you filter out poses, the bones will get locked in to a static coordinate frame and other weirdness.

Sadly, this is not documented anywhere in the Magic Leap SDK,